File: Exercises/rb101-rb109_small_problems/easy_7/lettercase_counter.rb
# frozen_string_literal: true

def letter_case_count(string)
  hash = {}
  hash[:lowercase] = string.count('a-z')
  hash[:uppercase] = string.count('A-Z')
  hash[:neither] = string.count('^a-zA-Z')
  hash
end

puts letter_case_count('abCdef 123') == { lowercase: 5, uppercase: 1, neither: 4 }
puts letter_case_count('AbCd +Ef') == { lowercase: 3, uppercase: 3, neither: 2 }
puts letter_case_count('123') == { lowercase: 0, uppercase: 0, neither: 3 }
puts letter_case_count('') == { lowercase: 0, uppercase: 0, neither: 0 }

File: ls215/exercises/small_problems/string_processing/lettercase_counter.js
function letterCaseCount(string) {
  const chars = string.split('');
  let counter = { lowercase: 0, uppercase: 0, neither: 0 };

  chars.forEach(char => {
    switch (true) {
      case /[a-z]/.test(char):
        counter.lowercase += 1;
        break;
      case /[A-Z]/.test(char):
        counter.uppercase += 1;
        break;
      default:
        counter.neither += 1;
    }
  });

  return counter;
}

console.log(letterCaseCount('abCdef 123'));  // { lowercase: 5, uppercase: 1, neither: 4 }
console.log(letterCaseCount('AbCd +Ef'));    // { lowercase: 3, uppercase: 3, neither: 2 }
console.log(letterCaseCount('123'));         // { lowercase: 0, uppercase: 0, neither: 3 }
console.log(letterCaseCount(''));            // { lowercase: 0, uppercase: 0, neither: 0 }
